File: external-src/com/thoughtworks/sql/Table.java
package com.thoughtworks.sql;

public class Table extends DBObject<Table> {

    protected Table(String expression) {
        super(expression);
    }

    public static Table table(String table) {
        return new Table(table);
    }

    @SuppressWarnings("nls")
    protected String fieldExpression(String fieldName) {
        if (hasAlias()) {
            return alias + "." + fieldName;
        }
        return expression+"."+fieldName;
    }
}

File: data/src/main/kotlin/org/tasks/data/db/Table.kt
package org.tasks.data.db

import org.tasks.data.sql.DBObject

class Table private constructor(private val name: String, alias: String?) : DBObject(name) {

    constructor(name: String) : this(name, null)

    fun column(column: String): Property = Property(this, column)

    override fun `as`(newAlias: String) = Table(name, newAlias)

    override fun toString(): String = alias?.let { "$expression AS $alias" } ?: expression

    fun name() = alias ?: name

    init {
        this.alias = alias
    }
}